src/nvim/os/shell.c

```diff
@@ -405,22 +405,23 @@ static bool out_data_decide_throttle(size_t size)
     return false;
   } else if (!visit) {
     started = os_hrtime();
-  } else if (visit % 20 == 0) {
+  } else {
     uint64_t since = os_hrtime() - started;
     if (since > (3 * NS_1_SECOND)) {
       received = visit = 0;
       return false;
     }
+    if (since < (visit * 0.1 * NS_1_SECOND)) {
+      return true;
+    }
   }
 
   visit++;
-  // Pulse "..." at the bottom of the screen.
-  size_t tick = (visit % 20 == 0)
-                ? 3  // Force all dots "..." on last visit.
-                : (visit % 4);
-  pulse_msg[0] = (tick == 0) ? ' ' : '.';
-  pulse_msg[1] = (tick == 0 || 1 == tick) ? ' ' : '.';
-  pulse_msg[2] = (tick == 0 || 1 == tick || 2 == tick) ? ' ' : '.';
+  // Pulse "..." at the bottom of the screen (4 states).
+  size_t tick = visit % 4;
+  pulse_msg[0] = (tick > 0) ? '.' : ' ';
+  pulse_msg[1] = (tick > 1) ? '.' : ' ';
+  pulse_msg[2] = (tick > 2) ? '.' : ' ';
```

Comment:
better, thanks. I think the comment can be unchanged though.

Reply:
ok.